### PR TITLE
Bug - [InferenceClient] - use proxy set in var env

### PR DESCRIPTION
**Pull request by morgandiverrez**
The AsyncInferenceClient doesn't use proxies set in environment variables. 
It uses aiohttp but doesn't set the trust_env parameter to True.

this PR handle trust_env parameter to and set aiohttp client and request client 
issue : [AsyncInferenceClient doesn't use proxies set in environment variables #2420](https://github.com/huggingface/huggingface_hub/issues/2420)

### CODE REVIEW

File: src/huggingface_hub/inference/_generated/_async_client.py [hunk 1]
```diff
@@ -149,6 +151,7 @@ def __init__(
         headers: Optional[Dict[str, str]] = None,
         cookies: Optional[Dict[str, str]] = None,
         proxies: Optional[Any] = None,
+        trust_env: bool = True,
```
Comment (Benvii) [inline src/huggingface_hub/inference/_generated/_async_client.py:154]: Breaking change, default value is False

```suggestion
        trust_env: bool = False, # Default value False documented here https://github.com/aio-libs/aiohttp/blob/v3.10.0/docs/client_reference.rst?plain=1#L195 
```

**Review by Benvii — CHANGES_REQUESTED** [review]
Prevent the breaking change for aiohttp default trust_env is False see my suggestion

File: src/huggingface_hub/inference/_generated/_async_client.py (hunk 1 above)
Comment (morgandiverrez) [reply src/huggingface_hub/inference/_generated/_async_client.py:154]: Yes, thank you, error during testing.

**Review by Benvii — APPROVED** [review]
Thanks @morgandiverrez, looks good to me 👍️.

I saw that @Wauplin worked on this files a few weeks ago, thanks in advance for taking a look at this small PR.
Really helpful fix for those working under corporate proxy that want to take advantage of existing environment variables.

**Comment by morgandiverrez** [comment]
hello @severo @XciD , is it possible to taking a look at this small PR, it's really helpful fix

**Comment by Wauplin** [comment]
@morgandiverrez @Benvii thanks for the PR and your inputs and sorry for the delay! I'm having a look at it right now :)

**Review by Wauplin — APPROVED** [review]
@morgandiverrez I hope you don't mind, I pushed commits to your PR. Changing stuff in `AsyncClientInference` without changing `InferenceClient` is a bit more complex/annoying as it requires to update a generating script. Anyway, now we should have something good. I didn't want to add `trust_env` to `InferenceClient` since this PR is mostly about `AsyncInferenceClient`. 

New commits:
- Do not modify `InferenceClient` https://github.com/huggingface/huggingface_hub/pull/2421/commits/263d64b276060f6b2e50329c290175ea72a27aec
- Add `trust_env` parameter + respect it in `AsyncInferenceClient` https://github.com/huggingface/huggingface_hub/pull/2421/commits/15ba44a715bd70451c6b039a32ef4ae2f7c1b3fb (also updating the generation script)
- (misc) add docstring + clean newlines https://github.com/huggingface/huggingface_hub/pull/2421/commits/cd47edbd81c3401a5d31d661b83314247fae1b49 https://github.com/huggingface/huggingface_hub/pull/2421/commits/392e96e93530d27798c4d91c960b59a2744b1114

Once the CI is green, I think we'll be good to merge.

**Comment by Wauplin** [comment]
Failing tests are unrelated so let's merge this :)

**Pull request merged**